docs/src/operate/security/http_local_authentication.md

```diff
@@ -0,0 +1,405 @@
+curl http://rpi4-dca632efb150:8001/c8y/inventory/managedObjects 
+```
+
+:::note
+The firewall of the box running the Cumumocity mapper has also to be configured
+to accept incoming requests on the port used by the Cumulocity HTTP Proxy (`8001` per default)
+:::
+
+## Enable HTTPS
+
+The next step is to enable HTTPS, the clients authenticating the servers.
+
+For that, one needs two certificates:
+- one for the main **tedge-agent** running the File Transfer Service
+- another for the Cumulocity **tedge-mapper** running the Cumulocity HTTP Proxy.
+
+These two certificates have also to be trusted by the clients, i.e. the child devices. 
+Hence, the signing certificate will have to be added to the list of trusted root certificates on each child device.
+
+### Generating Certificates
+
+Thin-edge currently provides no specific tool to generate and deploy certificates over child devices.
+
+You can use the following script to generate all the required certificates.
+
+First, one needs a signing key that will be used to sign the certificates of the **tedge-agent** and **tedge-mapper**:
+
+```sh
+DEVICE=$(tedge config get device.id)
+
+## Signing certificate
+openssl req \
+    -new \
+    -x509 \
+    -days 100 \
+    -extensions v3_ca \
+    -nodes \
+    -subj "/O=thin-edge/OU=$DEVICE/CN=tedge-ca" \
+    -keyout tedge-local-ca.key \
+    -out tedge-local-ca.crt
+```
+
+This signing certificate has to be trusted by the main device as well as all the child devices:
+
+```sh title="on the main as well as the child devices"
+cp tedge-local-ca.crt /usr/local/share/ca-certificates
+sudo update-ca-certificates
+```
+
+One can then proceed with a certificate for the **tedge-agent** of the main device:
+
+```sh
+## main agent private key
+openssl genrsa -out main-agent.key 2048
+
+## main agent certificate signing request
+openssl req -out main-agent.csr \
+    -key main-agent.key \
+    -subj "/O=thin-edge/OU=$DEVICE/SN=main-agent/CN=localhost" \
+    -new
+
+## signing the main agent certificate
+cat > v3.ext << EOF
+authorityKeyIdentifier=keyid
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, keyAgreement
+extendedKeyUsage = serverAuth, clientAuth
+subjectAltName=DNS:$(hostname),DNS:localhost
+EOF
+
+openssl x509 -req \
+    -in main-agent.csr \
+    -CA tedge-local-ca.crt \
+    -CAkey tedge-local-ca.key \
+    -extfile v3.ext \
+    -CAcreateserial \
+    -out main-agent.crt \
+    -days 100
+```
+
+And another one for the **tedge-mapper**:
+
+```sh
+## mapper private key
+openssl genrsa -out c8y-mapper.key 2048
+
+## mapper certificate signing request
+openssl req -out c8y-mapper.csr -key c8y-mapper.key \
+    -subj "/O=thin-edge/OU=$DEVICE/SN=c8y-mapper/CN=localhost" \
+    -new
+
+## signing the mapper certificate
+cat > v3.ext << EOF
+authorityKeyIdentifier=keyid
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, keyAgreement
+extendedKeyUsage = serverAuth, clientAuth
+subjectAltName=DNS:$(hostname),DNS:localhost
```

Comment:
I wonder whether it's worth clarifying how this works in case people wish to use an IP address to connect instead. Firstly, it's not unlikely that people will want to connect by something other than localhost or the hostname of the machine, Secondly, when using an IP it's easy to fall into the trap of using e.g. `DNS:192.168.0.25` where this should actually be `IP:192.168.0.25` (and if you make mistakes here, you'll get next to no help in the error message to enlighten you as to what is wrong).

Reply:
Added a note